src/python/pants/engine/rules.py

```diff
@@ -244,6 +253,8 @@ def rule_decorator(func, **kwargs) -> Callable:
         pants_rules[effective_name] = func
     else:
         prev_func = pants_rules[effective_name]
+        # TODO: Remove this check? We also check for uniqueness of effective_name in Rust code,
+        #  and that check is more comprehensive, as it also covers intrinsics.
```

Comment:
This catches the case where two rules with the same name are declared in a module (which Python would otherwise allow). In that case, only one would actually end up accessible in the module, and the Rust-side validation wouldn't catch it.

Reply:
Right, I have removed this TODO.